src/climate_learn/models/module.py

```diff
@@ -0,0 +1,103 @@
+# Standard library
+from typing import Any, Callable, Dict, List, Optional
+
+# Third party
+import torch
+from torch.optim.lr_scheduler import _LRScheduler as LRScheduler
+import pytorch_lightning as pl
+
+
+class LitModule(pl.LightningModule):
+    def __init__(
+        self,
+        net: torch.nn.Module,
+        optimizer: torch.optim.Optimizer,
+        lr_scheduler: LRScheduler,
+        train_loss: Callable,
+        val_loss: List[Callable],
+        test_loss: List[Callable],
+        train_target_transform: Optional[Callable] = None,
+        val_target_transforms: Optional[List[Callable]] = None,
+        test_target_transforms: Optional[List[Callable]] = None,
+    ):
+        super().__init__()
+        self.net = net
+        self.optimizer = optimizer
+        self.lr_scheduler = lr_scheduler
+        self.train_loss = train_loss
+        self.val_loss = val_loss
+        self.test_loss = test_loss
+        self.train_target_transform = train_target_transform
+        self.val_target_transform = val_target_transforms
+        self.test_target_transform = test_target_transforms
+
+    def forward(self, x):
+        return self.net(x)
+
+    def training_step(self, batch: Any, batch_idx: int):
+        x, y, in_variables, out_variables = batch
+        yhat = self(x).to(device=y.device)
```

Comment:
This is necessary because of the models which PyTorch Lightning would not autocast such as climatology. Climatology is set before casting to device happens.